src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c

```diff
@@ -415,19 +436,24 @@ free_berdata(struct berval **array)
     }
 }
 
-/* Decoding ASN.1 encoded key */
-static struct berval **
-krb5_encode_krbsecretkey(krb5_key_data *key_data_in, int n_key_data,
-                         krb5_kvno mkvno) {
-    struct berval **ret = NULL;
-    int currkvno;
-    int num_versions = 1;
-    int i, j, last;
+/*
+ * Encode krb5_key_data into a berval struct for ldap insertion.
+ */
+static krb5_error_code
+encode_keys(krb5_key_data *key_data_in, int n_key_data,
+                 krb5_kvno mkvno, struct berval **ret_out)
```

Comment:
Please rename "ret_out" to "bval_out" and "ret" to "bval".  I'm not super-attached to "bval", but "ret" doesn't make sense when we aren't actually returning it as we were in the old code, and "ret_out" is just confusing.